fern/docs-config-v2/definition/docs.yml

```diff
@@ -0,0 +1,310 @@
+      border:
+        type: optional<ColorConfig>
+        docs: Defaults to white/10 in dark mode, and black/10 in light mode
+
+  ColorConfig:
+    properties:
+      dark: optional<string>
+      light: optional<string>
+
+  ImageConfig:
+    properties:
+      dark: optional<string>
+      light: optional<string>
+    docs: Path to the image.
+
+  HeaderConfig:
+    properties:
+      position:
+        type: optional<HeaderPosition>
+        docs: Defaults to "sticky"
+      render-mode: optional<HeaderRenderMode>
+      header-height-rem:
+        type: optional<double>
+        docs: Defaults to 4rem (64px)
+      border-radius-rem:
+        type: optional<double>
+        docs: If set, the header will have rounded corners, and a border around the entire header.
+      links: optional<list<HeaderLink>>
+
+  HeaderRenderMode:
+    union:
+      border-bottom: HeaderRenderModeBorderBottom
+      rounded: HeaderRenderModeRounded
+      no-border: HeaderRenderModeNoBorder
+
+  HeaderRenderModeBorderBottom:
+    properties:
+      border-thickness-px:
+        type: optional<integer>
+        docs: Defaults to 1px
+      border-color: optional<ColorConfig>
+
+  HeaderRenderModeRounded:
+    docs: |
+      The header will have a border around the entire header, and its corners will be rounded.
+    properties:
+      border-radius-rem:
+        type: optional<double>
+        docs: Defaults to 0.5rem (8px)
+      border-thickness-px:
+        type: optional<integer>
+        docs: Defaults to 1px
+      border-color: optional<ColorConfig>
+      shadow-size:
+        type: optional<ShadowSize>
+        docs: Defaults to none
+
+  ShadowSize:
+    enum:
+      - none
+      - small
+      - default
+      - medium
+      - large
+      - xlarge
+      - name: xxlarge
+        value: 2xlarge
+
+  HeaderRenderModeNoBorder:
+    properties: {}
+
+  HeaderPosition:
+    union:
+      sticky: HeaderStickyBackdrop
+      static: HeaderStaticBackdrop
+
+  HeaderStickyBackdrop:
+    union:
+      solid: HeaderBackdropSolid
+      blur: HeaderBackdropBlur
+    docs: Defaults to blur
+
+  HeaderStaticBackdrop:
+    union:
+      solid: HeaderBackdropSolid
+    docs: Defaults to transparent
+
+  HeaderBackdropTransparent:
+    properties: {}
+
+  HeaderBackdropSolid:
+    properties:
+      background-color:
+        type: optional<ColorConfig>
+        docs: Defaults to colors.foreground
+
+  HeaderBackdropBlur:
+    properties:
+      backdrop-blur-rem:
+        type: optional<double>
+        docs: Defaults to 2rem (32px).
+      backdrop-blur-color:
+        type: optional<ColorConfig>
+        docs: Defaults to transparent
+
+  HeaderLink:
+    properties:
+      render: HeaderLinkRender
+      url: string
+      text: string
+
+  HeaderLinkRender:
+    union:
+      primary: HeaderLinkRenderPrimary
+      secondary: HeaderLinkRenderSecondary
+
+  HeaderLinkRenderPrimary:
+    properties: {}
+
+  HeaderLinkRenderSecondary:
+    properties: {}
```

Comment:
```yaml
header: 
  links: 
    url: ...
    text: ...
    render: 
      type: primary
```
is probably worse than 

```
header: 
  links: 
    url: ...
    text: ...
    style: primary # my guess is style is better than render? 
```

To model this you can do something like
```
HeaderLink: 
  base-properties: 
    url: string
    text: string
  discriminant: style
  union: 
     primary: {}
     secondary: {}
```
or you can keep it as is and introduce enums for `primary` and `secondary`